api/src/main/java/io/strimzi/api/kafka/AbstractCustomResource.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.api.kafka;
+
+import io.fabric8.kubernetes.client.CustomResource;
+import io.strimzi.api.kafka.model.Spec;
+import io.strimzi.api.kafka.model.status.HasStatus;
+import io.strimzi.api.kafka.model.status.Status;
+
+public abstract class AbstractCustomResource<T extends Spec, S extends Status> extends CustomResource implements HasStatus<S> {
```

Comment:
Why does this have to be a class extending `CustomerResource` rather than an interface that extends `HasMetadata` and `HasStatus`?

Reply:
Agreed with tom here,  seems like making  a combined interface would probably be a marginally better solution here